src/main/java/net/openhft/chronicle/bytes/UncheckedNativeBytes.java

```diff
@@ -908,6 +914,59 @@ public Bytes<U> appendUtf8(char[] chars, @NonNegative int offset, @NonNegative i
         return this;
     }
 
+    @Override
+    public @NotNull UncheckedNativeBytes<U> append(double d)
+            throws BufferOverflowException, IllegalStateException {
+        if (!Decimalizer.INSTANCE.toDecimal(d, this))
+            append(Double.toString(d));
+        return this;
+    }
+
+
+    @Override
+    public @NotNull UncheckedNativeBytes<U> append(float f)
+            throws BufferOverflowException, IllegalStateException {
+        if (!Decimalizer.INSTANCE.toDecimal(f, this))
+            append(Float.toString(f));
+        return this;
+    }
+
+    @Override
+    public void append(boolean negative, long mantissa, int exponent) {
+        ensureCapacity(48);
+        if (negative)
+            rawWriteByte((byte) '-');
+        long pos = writePosition();
+        if (exponent <= 0) {
+            rawWriteByte((byte) '0');
+            rawWriteByte((byte) '.');
+            while (exponent++ < 0)
+                rawWriteByte((byte) '0');
+            exponent = -1;
+        }
+
+        do {
+            if (exponent-- == 0)
+                rawWriteByte((byte) '.');
+            long base = mantissa % 10;
+            mantissa /= 10;
+            rawWriteByte((byte) ('0' + base));
+        } while (mantissa > 0 || exponent >= 0);
+        reverseBytesFrom(pos);
+    }
+
+    protected void reverseBytesFrom(long pos) {
+        long pos2 = writePosition()-1;
+        while (pos2 > pos) {
+            byte b1 = bytesStore.readByte(pos);
+            byte b2 = bytesStore.readByte(pos2);
+            bytesStore.writeByte(pos, b2);
+            bytesStore.writeByte(pos2, b1);
+            pos++;
+            pos2--;
+        }
+    }
```

Comment:
Could this also live in decimalizer? it seems to be repeated in AbstractBytes and UncheckedNativeBytes

Reply:
It repeats the same structure but using it in Decimaliser would turn every byte into a polymorphic call.
There is a benchmark so we could explore how much difference it makes.